src/components/application_manager/test/commands/mobile/show_test.cc

```diff
@@ -590,6 +636,90 @@ TEST_F(ShowRequestTest, Run_MainField4_WrongSyntax) {
+  const size_t num_tags = 1;
+  int32_t tags[num_tags] = {hmi_apis::Common_MetadataType::mediaArtist};
+  TestSetupHelperWithMetadata(msg,
+                              hmi_apis::Common_TextFieldName::mainField1,
+                              am::strings::main_field_1,
+                              num_tags,
+                              tags);
+  command->Run();
+}
+
+TEST_F(ShowRequestTest, Run_MainField1_MultipleMetadataTags) {
+  MessageSharedPtr msg = CreateMsgParams();
+
+  SharedPtr<ShowRequest> command(CreateCommand<ShowRequest>(msg));
+
+  text_field_ = "Main_Field_1";
+  const size_t num_tags = 5;
+  int32_t tags[num_tags] = {hmi_apis::Common_MetadataType::mediaTitle,
+                            hmi_apis::Common_MetadataType::mediaArtist,
+                            hmi_apis::Common_MetadataType::rating,
+                            hmi_apis::Common_MetadataType::humidity,
+                            hmi_apis::Common_MetadataType::currentTemperature};
+  TestSetupHelperWithMetadata(msg,
+                              hmi_apis::Common_TextFieldName::mainField1,
+                              am::strings::main_field_1,
+                              num_tags,
+                              tags);
+  command->Run();
+}
+
+TEST_F(ShowRequestTest, Run_MainField2_MetadataTag) {
+  MessageSharedPtr msg = CreateMsgParams();
+
+  SharedPtr<ShowRequest> command(CreateCommand<ShowRequest>(msg));
+
+  text_field_ = "Main_Field_2";
+  const size_t num_tags = 1;
+  int32_t tags[num_tags] = {hmi_apis::Common_MetadataType::mediaArtist};
+  TestSetupHelperWithMetadata(msg,
+                              hmi_apis::Common_TextFieldName::mainField2,
+                              am::strings::main_field_2,
+                              num_tags,
+                              tags);
+  command->Run();
+}
+
+TEST_F(ShowRequestTest, Run_MainField3_MetadataTag) {
+  MessageSharedPtr msg = CreateMsgParams();
+
+  SharedPtr<ShowRequest> command(CreateCommand<ShowRequest>(msg));
+
+  text_field_ = "Main_Field_3";
+  const size_t num_tags = 1;
+  int32_t tags[num_tags] = {hmi_apis::Common_MetadataType::mediaArtist};
+  TestSetupHelperWithMetadata(msg,
+                              hmi_apis::Common_TextFieldName::mainField3,
+                              am::strings::main_field_3,
+                              num_tags,
+                              tags);
+  command->Run();
+}
+
+TEST_F(ShowRequestTest, Run_MainField4_MetadataTag) {
+  MessageSharedPtr msg = CreateMsgParams();
+
+  SharedPtr<ShowRequest> command(CreateCommand<ShowRequest>(msg));
+
+  text_field_ = "Main_Field_4";
+  const size_t num_tags = 1;
+  int32_t tags[num_tags] = {hmi_apis::Common_MetadataType::mediaArtist};
+  TestSetupHelperWithMetadata(msg,
+                              hmi_apis::Common_TextFieldName::mainField4,
+                              am::strings::main_field_4,
+                              num_tags,
+                              tags);
+  command->Run();
+}
+
```

Comment:
Should probably include a test for the case where metadata is included without the original field (`WARNINGS` response case).